File: app/src/main/java/io/github/ovso/righttoknow/ui/main/violationfacility/ViolationFacilityPresenter.java
package io.github.ovso.righttoknow.ui.main.violationfacility;

import android.content.Context;
import android.os.Bundle;
import androidx.annotation.IdRes;
import androidx.annotation.StringRes;
import io.github.ovso.righttoknow.framework.adapter.BaseAdapterDataModel;
import io.github.ovso.righttoknow.ui.main.violationfacility.model.VioFac;

/**
 * Created by jaeho on 2017. 8. 1
 */

public interface ViolationFacilityPresenter {

  void onActivityCreated(Bundle savedInstanceState);

  void setAdapterModel(BaseAdapterDataModel<VioFac> adapterDataModel);

  void onRecyclerItemClick(VioFac vioFac);

  void onRefresh();

  void onSearchQuery(String query);

  void onOptionsItemSelected(@IdRes int itemId);

  void onDestroyView();

  interface View {

    void setRecyclerView();

    void setAdapter();

    void refresh();

    void showLoading();

    void hideLoading();

    void navigateToViolationFacilityDetail(String webLink, String address);

    void setListener();

    Context getActivity();

    void setSearchResultText(@StringRes int resId);

    void showMessage(@StringRes int error_server);
  }
}

File: app/src/main/java/io/github/ovso/righttoknow/ui/main/violationfacility/ViolationFacilityPresenter.kt
package io.github.ovso.righttoknow.ui.main.violationfacility

import android.os.Bundle
import androidx.annotation.IdRes
import androidx.annotation.StringRes
import io.github.ovso.righttoknow.framework.adapter.BaseAdapterDataModel
import io.github.ovso.righttoknow.ui.main.violationfacility.model.VioFac

interface ViolationFacilityPresenter {
  fun onActivityCreated(savedInstanceState: Bundle?)
  fun setAdapterModel(adapterDataModel: BaseAdapterDataModel<VioFac>)
  fun onRecyclerItemClick(vioFac: VioFac)
  fun onRefresh()
  fun onSearchQuery(query: String)
  fun onOptionsItemSelected(@IdRes itemId: Int)
  fun onDestroyView()
  interface View {
    fun setRecyclerView()
    fun setAdapter()
    fun refresh()
    fun showLoading()
    fun hideLoading()
    fun navigateToViolationFacilityDetail(
      webLink: String?,
      address: String?
    )

    fun setListener()
    fun setSearchResultText(@StringRes resId: Int)
    fun showMessage(@StringRes error_server: Int)
  }
}